### DISCUSSION TITLE
Import-CallQueue -> Error 401

### DISCUSSION THREAD

**Discussion by donky-bison** [OfficeDev/TACO#19] [Q&A] [answered]
Hi 
I'm very excited about this app.
After deploying the app, I'm still facing problems using Import-CallQueue
- Using Power Automate I always get 401, even when authentication is set to "allow unauthenticatexd requests"
- Starting Import-callqueue direct in Azure Function works
- Can not use any other azure funtkion in Power Apps

Can you help me?

**Comment by JohanVeldhuis** [comment]
Hi @donky-bison,

It looks like that in that case the authentication piece might not be correctly configured.  When you are running the workflow manually can you tell me which step exactly fails? What I suspect is that this might be an issue getting the secrets from the Azure Key vault however to confirm that I will need to understand which step exactly fails.

In the meantime you can validate the permissions on the Azure Keyvault by performing the following steps:

- go to the **Azure portal**
- search for the **Key Vault** used by the App
- select **Access policies**
- make sure the **service account** is listed
- validate the permissions assigned to the account are at least the **get** permissions

In addition to this validate the following secrets are listed on the Secrets section:

- TACO-appSecret
- Teams-service-account-name
- Teams-service-account-secret
- 
Regards,
Johan

**Comment by donky-bison** [comment]
Hi

Thank you for your help.

The Access policies looks like that and "2 selected means" Get and List

***@***.***

Here are the defined secrets

***@***.***

Which steps failed running the workflow manually?

***@***.***


Best regards - Urs Egli

Von: Johan Veldhuis ***@***.***>
Gesendet: Freitag, 19. August 2022 16:38
An: OfficeDev/TACO ***@***.***>
Cc: Egli Urs ***@***.***>; Mention ***@***.***>
Betreff: Re: [OfficeDev/TACO] Imkport-CallQueue -> Error 401 (Discussion #19)


Hi @donky-bison<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdonky-bison&data=05%7C01%7Curs.egli%40bithawk.ch%7Cb3ba7202d61d494c10f608da81f05fda%7Cb58f94a366ad4b34bd9e7d8af8addb7d%7C0%7C0%7C637965166652613980%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=JiHk5wiHxVpJ4rxOZvypuNB0nCyb9V6cgStGghAPUZo%3D&reserved=0>,

It looks like that in that case the authentication piece might not be correctly configured. When you are running the workflow manually can you tell me which step exactly fails? What I suspect is that this might be an issue getting the secrets from the Azure Key vault however to confirm that I will need to understand which step exactly fails.

In the meantime you can validate the permissions on the Azure Keyvault by performing the following steps:

  *   go to the Azure portal
  *   search for the Key Vault used by the App
  *   select Access policies
  *   make sure the service account is listed
  *   validate the permissions assigned to the account are at least the get permissions

In addition to this validate the following secrets are listed on the Secrets section:

  *   TACO-appSecret
  *   Teams-service-account-name
  *   Teams-service-account-secret
  *

Regards,
Johan

-

**Reply by JohanVeldhuis** [reply]
Hi Urs,

Can you post a screenshot of the workflow failed so we can check which part failed?

Johan

**Comment by donky-bison** [comment]
Hi Johan

Is this the screenshot you requested??

***@***.***

Best regards - Urs Egli

Von: Johan Veldhuis ***@***.***>
Gesendet: Montag, 22. August 2022 08:58
An: OfficeDev/TACO ***@***.***>
Cc: Egli Urs ***@***.***>; Mention ***@***.***>
Betreff: Re: [OfficeDev/TACO] Imkport-CallQueue -> Error 401 (Discussion #19)


Hi Urs,

Can you post a screenshot of the workflow failed so we can check which part failed?

Johan

-

**Reply by JohanVeldhuis** [reply]
@donky-bison no but probably I should have been more clear. Please perform the steps below:

- go to make.powerapps.com
- in the **left menu** select flow
- search for the **import all call queues** workflow
- click on the **3 dots** and select **Run History**
- find the latest run and click on the date/time to show the results
- find the object with the **red cross**
- expand the object completely and share the screenshot (WARNING: prior to sharing the screenshot make sure the value data is hidden by masking it)

Johan

**Comment by donky-bison** [comment]
Hi

I hope this is ok

***@***.***


***@***.***


Best regards - Urs Egli

Von: Johan Veldhuis ***@***.***>
Gesendet: Montag, 22. August 2022 15:08
An: OfficeDev/TACO ***@***.***>
Cc: Egli Urs ***@***.***>; Mention ***@***.***>
Betreff: Re: [OfficeDev/TACO] Imkport-CallQueue -> Error 401 (Discussion #19)


@donky-bison<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdonky-bison&data=05%7C01%7Curs.egli%40bithawk.ch%7C598ad8642b6d4a08fb4c08da843f6700%7Cb58f94a366ad4b34bd9e7d8af8addb7d%7C0%7C0%7C637967705100595488%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=inv2KhIsqd4ypUdY3g3eQqugUHVby4DnqYEeOyGJtpQ%3D&reserved=0> no but probably I should have been more clear. Please perform the steps below:

  *   go to make.powerapps.com
  *   in the left menu select flow
  *   search for the import all call queues workflow
  *   click on the 3 dots and select Run History
  *   find the latest run and click on the date/time to show the results
  *   find the object with the red cross
  *   expand the object completely and share the screenshot (WARNING: prior to sharing the screenshot make sure the value data is hidden by masking it)

Johan

-

**Reply by JohanVeldhuis** [reply]
@donky-bison it is probably me but I don't see a screenshot in this post

**Comment by donky-bison** [comment]
Hi 

It should work know to see the screenshots

![image](https://user-images.githubusercontent.com/14820231/186149218-7c931131-ab02-40a7-85b4-6107c02b74b4.png)
![image](https://user-images.githubusercontent.com/14820231/186149300-17eba9dd-8c6c-4266-9532-8dbde915e3fc.png)


best regards

**Reply by JohanVeldhuis** [reply]
@donky-bison looks like the Azure Function url is not configured correctly. Please perform the following steps:

- go to make.powerapps.com
- select **solutions** from the left menu
- find the **TACO** app,select the **3 dots** and select **Edit**
- in the list search for **VAR - FunctionApp**, select the **3 dots** and select **Edit**
- validate the value specified in the **Current Value** field, for example if your Azure Function URL is https://azfunction.azurewebsites.net only use the host part, in this case **azfunction**
- after making the change click save
- run the **import all call queues** flow again

Johan

**Reply by donky-bison** [reply]
**import all autoattendant**
It seems to work now. I had to create a second AA otherwise I got an error in parsing Json - see grafic
![image](https://user-images.githubusercontent.com/14820231/186346302-bf737bdc-cf3a-4035-a746-33027d45fc2b.png)

**import all call queues**
error in parsing Json - see grafik 2 and 3
![image](https://user-images.githubusercontent.com/14820231/186346856-651ebd72-7bb3-4105-80f4-955d2cb353e4.png)
![image](https://user-images.githubusercontent.com/14820231/186346966-f97f0fef-3dd5-43c6-a957-1edf2d53c7e6.png)

Many thanks for your help

**Reply by JohanVeldhuis** [reply]
@donky-bison good to hear it resolved your issue. We didn't thought about the scenario that there is only one AA or CQ so I will add that to our to do list to update the workflow to take this into account.

For the CQ issue please replace the JSON in the workflow with the one below:

{

    "type": "array",

    "items": {

        "type": "object",

        "properties": {

            "Name": {

                "type": "string"

            },

            "AgentAlertTime": {},

            "OverflowThreshold": {},

            "OverflowAction": {},

            "OverflowActionTarget": {},

            "OverflowSharedTextToSpeechPrompt": {},

            "OverflowSharedVoicemailAudioFilePromptFileName": {},

            "TimeoutThreshold": {},

            "TimeoutAction": { },

            "TimeoutActionTarget": {},

            "TimeoutSharedTextToSpeechPrompt": {},

            "TimeoutSharedVoicemailAudioFilePromptFileName": {},

            "UseDefaultWelcomeMusic": {},

            "WelcomeMusicFileName": {},

            "UseDefaultMusicOnHold": {},

            "MusicOnHoldFileName": {}

        },

        "required": [

            "Name"

        ]

    }

}

Regards,
Johan

**Reply by donky-bison** [reply]
it works - thanks

**Reply by JohanVeldhuis** [reply]
@donky-bison good to hear it resolved the issue I have created an issue to make sure we fix this in the next version of the app.

**Comment by JohanVeldhuis** [comment, marked as answer]
Solution for this issue was:

- make sure the variable  VAR - FunctionApp
- updated the JSON structure in the Import All Call Queues work flow --> bug is created
